Description: Return NOTSUPP for recognized GET_DIR_DELEGATION
 Backport only the two dispatch fields from the accepted upstream correction.
 The optional operation remains unimplemented; preserve its operation number.
Author: Suhas Athani <Suhas.Athani@ibm.com>
Origin: upstream, https://github.com/nfs-ganesha/nfs-ganesha/commit/98eb77c4fd94effa6882b214698a77752ec79f6e
Bug: https://github.com/nfs-ganesha/nfs-ganesha/issues/1385
Forwarded: not-needed
Last-Update: 2026-10-14
--- a/src/Protocols/NFS/nfs4_Compound.c
+++ b/src/Protocols/NFS/nfs4_Compound.c
@@ -438,9 +438,9 @@
 		.exp_perm_flags = 0},
 	[NFS4_OP_GET_DIR_DELEGATION] = {
 		.name = "OP_GET_DIR_DELEGATION",
-		.funct = nfs4_op_illegal,
+		.funct = nfs4_op_notsupp,
 		.resume = nfs4_default_resume,
-		.free_res = nfs4_op_illegal_Free,
+		.free_res = nfs4_op_notsupp_Free,
 		.resp_size = sizeof(GET_DIR_DELEGATION4res),
 		.exp_perm_flags = 0	/* tbd */},
 	[NFS4_OP_GETDEVICEINFO] = {
